### PR TITLE
refactor: Move common test helpers to core

### PR DESCRIPTION
**Pull request by tboldagh**
This PR moves MultiTrajectory test helpers to the Core package. Thanks to that we would be able to reuse them in ATLAS to setup the compatible test. Once changes in MTJ implemenation in ACTS change it would be reflected in the test and during integration of that change the shared test will indicate what exacly needs to be updated.

@paulgessinger @CarloVarni - since you two discussed that a bit @ workshop please let me know if this is the right direction. As you see I have left out thet TestTrackProxy and TestSourceLink in order minimize polution of Core package.

### CODE REVIEW

File: Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp [hunk 1]
```diff
@@ -88,48 +88,46 @@ struct TestTrackState {
...
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Jacobian)) {
+      ts.jacobian() = jacobian;
+    }
+    ts.chi2() = chi2;
+    ts.pathLength() = pathLength;
+    // source link defines the uncalibrated measurement
+    ts.setUncalibratedSourceLink(Acts::SourceLink{sourceLink});
+    // create calibrated measurements from source link
+    if (ACTS_CHECK_BIT(mask, TrackStatePropMask::Calibrated)) {
+      testSourceLinkCalibrator<trajectory_t>(Acts::GeometryContext{}, ts);
```
Comment (paulgessinger) [inline Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp:128]: The calibration context goes away?

**Review by paulgessinger — COMMENTED** [review]
I think this is it! Thanks @tboldagh

File: Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp (hunk 1 above)
Comment (tboldagh) [reply Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp:128]: I just made fillTrackState a method of TestTrackState. 
The sourceLink related code is still not compiling though. 
Wanted to hear about if the direction is right one.

Comment (tboldagh) [reply Tests/CommonHelpers/Acts/Tests/CommonHelpers/TestTrackState.hpp:128]: I understand now. This is an effect of git trying to resolve the conflict. Adding back all needed args.

**Comment by tboldagh** [comment]
@paulgessinger  - would moving GenerateParameters.h to Core be still ok?

**Comment by tboldagh** [comment]
I am not able to reproduce the compilation error enabling these: `-DACTS_BUILD_EVERYTHING=ON ACTS_BUILD_PLUGIN_PODIO=ON`

Any hint is welcome.

**Comment by tboldagh** [comment]
Hmm, the test coverage is smaller. But how is that possible after just moving code around?
Can it be that because TestSourceLink has not some more of the code in the header it needs to be checked?

**Comment by paulgessinger** [comment]
@tboldagh The coverage calculation is not 100% robust. If it's a small change and we don't think it's damaging, we can ignore it.

**Review by paulgessinger — COMMENTED** [review]
Changes overall look clean. What do you think about putting the files into a subfolder `detail_test` to indicate they're not really meant for **public** consumption?

File: Core/include/Acts/EventData/TestSourceLink.hpp
[the remark this answers is not in the document]
Comment (tboldagh) [reply Core/include/Acts/EventData/TestSourceLink.hpp:1]: Do you mean extra folder in the EventData? Or next to EventData? 
What about calling it "RuntimeConceptCheckHelpers" (need better name).   
i.e. the test cases that are there are just a method to validate the conformity of whatever implementaiton delivers matches to the requrements. It is actually an intersting s/w engennering problem.

[the remark this answers is not in the document]
Comment (tboldagh) [reply Core/include/Acts/EventData/TestSourceLink.hpp:1]: @paulgessinger - please see the changes made

File: Tests/CommonHelpers/Acts/Tests/CommonHelpers/MeasurementsCreator.hpp [hunk 2]
```diff
@@ -26,6 +26,7 @@
 namespace Acts {
 namespace Test {
 
+using namespace Acts::detail::Test;
```
Comment (paulgessinger) [inline Tests/CommonHelpers/Acts/Tests/CommonHelpers/MeasurementsCreator.hpp:29]: `using namespace` is dangerous. Can this be avoided?

Comment (tboldagh) [reply Tests/CommonHelpers/Acts/Tests/CommonHelpers/MeasurementsCreator.hpp]: I indeed uncovered hidden depency on this namespace in several source files. Fixed now.

**Review by paulgessinger — APPROVED** [review]

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [b533a876f00c031ab019a4f098b63a76048f51ad]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534782) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534781) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534780) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534779) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534778) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534777) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534776) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534775) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534774) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534773) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534772) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534771) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534770) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534769) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534768) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534767) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534766) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534765) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534764) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534763) |
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534762) |
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34534761) |

**Pull request merged**